src/celesto/auth.py

```
@@ -38,38 +43,175 @@ def _resolve_base_url(base_url: str | None = None) -> str:
-        api_key,
-    )
+    account = _credential_account(resolved_base_url)
+    try:
+        keyring.set_password(
+            KEYRING_SERVICE,
+            account,
+            api_key,
+        )
+        _set_credential_store_preference(account, "keyring")
+        return "keyring"
+    except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)